### PR TITLE
ffix: subwallet account does not work

### LINKED ISSUE #2590: Apparently Subwallet doesn't work on Connect
Apparently, Subwallet doesn't work on Connect, let's crack on this why in another issue to push this forward :)

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/2574#issuecomment-1072989339_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2590
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [x] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image

### CODE REVIEW

**Comment by vikiival** [comment]
Screenshot 2022-03-20 at 11 35 14

Still the same :/

**Comment by vikiival** [comment]
https://github.com/kodadot/nft-gallery/issues/2590#issuecomment-1073221435

**Comment by yangwao** [comment]
👀

**Comment by Jarsen136** [comment]
> [#2590 (comment)](https://github.com/kodadot/nft-gallery/issues/2590#issuecomment-1073221435)

hello, please check my comment below.
https://github.com/kodadot/nft-gallery/issues/2590#issuecomment-1073225820

File: components/common/WalletModal.vue [hunk 1]
```diff
@@ -125,7 +126,7 @@ export default class WalletModal extends Vue {
     }
   }
 
-  protected setWallet(wallet: Wallet): void {
+  protected async setWallet(wallet: BaseDotsamaWallet): Promise<void> {
```
Comment (roiLeo) [inline components/common/WalletModal.vue:129]: you use async here but there is no await in method
why did you change the type? in future we want to authorize not only dotsama wallet

File: components/common/WalletModal.vue [hunk 2]
```diff
@@ -144,6 +145,18 @@ export default class WalletModal extends Vue {
...
+
+      // init account
+      wallet
+        .getAccounts()
+        .then((data) => {
+          this.walletAccounts = data ?? []
+        })
+        .catch((e) => {
+          console.error('init account error', e)
+        })
```
Comment (roiLeo) [inline components/common/WalletModal.vue:157]: same thing is done in `wallet.subscribeAccounts`

**Review by roiLeo — COMMENTED** [review]
Hey! I don't think this modification will changes anything

File: components/common/WalletModal.vue (hunk 2 above)
Comment (Jarsen136) [reply components/common/WalletModal.vue:157]: However, in subwallet, `wallet.subscribeAccounts` will not do the same thing. Therefore the address list could not  be shown.

File: components/common/WalletModal.vue (hunk 1 above)
Comment (Jarsen136) [reply components/common/WalletModal.vue:129]: you are right. I will remove it

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 165.46 USD/KSM ~ 0.302 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x700cf7eebd0c61f7712a6d80e5d845332331a6a88d5a9b9baf8dc673df4d4d4c](https://kusama.subscan.io/extrinsic/0x700cf7eebd0c61f7712a6d80e5d845332331a6a88d5a9b9baf8dc673df4d4d4c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**